### PR TITLE
retry process jobs

### PR DESCRIPTION
**Pull request by justinfisk**
Not 100% sure this is ready for PR, but want someone to review...

This does 2 things:
1. adapts the fetch retry code to also retry failed process jobs (#266)
2. removes the postgresql-specific `distinct` call. 

I believe the other `distinct` calls (in `inventory.dbinv.api`) are not postgresql-specific because they do not take a column as an argument.

### CODE REVIEW

File: gips/datahandler/scheduler.py [hunk 1]
```diff
@@ -73,62 +74,91 @@ def schedule_query ():
...
-            Logger().log('Attempting to reschedule incomplete tasks')
-
-            retries = []
-            fails = []
-            for a in needs_cleanup:
-                if a.num_retries >= 3:
-                    fails.append(a)
-                    # if it has been retried 3 times, give up
-                    a.status = 'failed'
+            print "j is", j
```
Comment (ags-tolson) [inline gips/datahandler/scheduler.py]: optional:  may be better to log() here instead of print

Comment (justinfisk) [inline gips/datahandler/scheduler.py]: Actually, would be better to do neither. Debug cruft. Logs relevant info later

File: gips/datahandler/scheduler.py [hunk 2]
```diff
@@ -73,62 +74,91 @@ def schedule_query ():
...
-                    .format(pformat([(f.id, f.driver, f.asset, f.tile, f.date)
-                                     for f in fails]))
-                )
-            if retries:
-                Logger().log(
-                    "Retrying fetch\n{}"
-                    .format(pformat([(r.id, r.driver, r.asset, r.tile, r.date)
-                                     for r in retries]))
-                )
+                    raise # TODO raise something useful
```
Comment (ags-tolson) [inline gips/datahandler/scheduler.py]: optional:  I like `RuntimeError('Unreachable line reached:  model was neither Product nor Asset')` similar to:

https://github.com/Applied-GeoSolutions/gips/blob/batch-scheduler/gips/datahandler/queue/queue.py#L43

File: gips/datahandler/scheduler.py (hunk 1 above)
Comment (ags-tolson) [inline gips/datahandler/scheduler.py]: sounds good to me; there is `log.debug()` if you want though.

**Comment by ags-tolson** [comment]
Provided I understand `(something).annotate(...)` I think it looks correct.

I know we're on a deadlin atm but if you want, I had a brainstorm to DRY out the conditional:

```diff
+                if model == dbinv.models.Asset:
+                    needs_cleanup = needs_cleanup.filter(
+                        assetstatuschange__status='requested',
+                    ).annotate(
+                        num_retries=models.Count('assetstatuschange__status'),
+                    )
+                elif model == dbinv.models.Product:
+                    needs_cleanup = needs_cleanup.filter(
+                        productstatuschange__status='requested',
+                    ).annotate(
+                        num_retries=models.Count('productstatuschange__status'),
+                    ) 
```

This way has fewer repeated lines:
```python
magic_str = {
    dbinv.models.Asset: 'assetstatuschange__status', 
    dbinv.models.Product: 'productstatuschange__status'}[model]
needs_cleanup = needs_cleanup.filter(
    **{magic_str: 'requested'}
).annotate(
    num_retries=models.Count(count_magic_str)
)
```

**Pull request closed without merging**